### PR TITLE
refactor: alternate consensus engines

### PR DESCRIPTION
**Pull request by reecepbcups**
## Summary

The relayer is tied heavily to CometBFT. To support Gordian w/ IBC, we need to generalize this. This adds a new interface that gives exactly what the relayer needs to satisfy CometBFT logic, without tying it directly to CometBFT client calls.

This way CometBFT can use the http RPC, while gordian can use its own http RPC or gRPC. 

## Future PRs
- Update to SDK v0.52 / IBC-Go v10
- add Gordian gRPC/RPC Support with simplified types
- Add gordian <> cometbft IBC test

### CODE REVIEW

File: relayer/chains/cosmos/tx.go [hunk 1]
```diff
@@ -1668,7 +1669,8 @@ func (cc *CosmosProvider) PrepareFactory(txf tx.Factory, signingKey string) (tx.
 		return tx.Factory{}, err
 	}
 
-	cliCtx := client.Context{}.WithClient(cc.RPCClient).
+	cliCtx := client.Context{}.
+		// WithClient(cc.RPCClient). // TODO(reece): will this break gordian? hwo does server/v2 handle this?
```
Comment (reecepbcups) [inline relayer/chains/cosmos/tx.go]: do we even need this if all test pass? or are we lacking coverage

File: relayer/chains/cosmos/feegrant.go [hunk 2]
```diff
@@ -332,7 +332,8 @@ func (cc *CosmosProvider) EnsureBasicGrants(ctx context.Context, memo string, ga
 	}
 
 	if len(msgs) > 0 {
-		cliCtx := client.Context{}.WithClient(cc.RPCClient).
+		cliCtx := client.Context{}.
+			// WithClient(cc.RPCClient). // TODO(reece): how does server/v2 handle this?
```
Comment (reecepbcups) [inline relayer/chains/cosmos/feegrant.go]: do we even need this if all test pass? or are we lacking coverage

File: relayer/chains/cosmos/tx.go (hunk 1 above)
Comment (reecepbcups) [reply relayer/chains/cosmos/tx.go]: talked with Justin on this, was added for feegrant. We are good to remove.

**Review by jtieri — APPROVED** [review]
everything looks good! this is a great refactor

**Pull request merged**